addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -7320,7 +7320,7 @@ registry.BackgroundPosition = SnippetOptionWidget.extend({
     backgroundType: function (previewMode, widgetValue, params) {
         this.$target.toggleClass('o_bg_img_opt_repeat', widgetValue === 'repeat-pattern');
         this.$target.css('background-position', '');
-        this.$target.css('background-size', '');
+        this.$target.css('background-size', widgetValue !== 'repeat-pattern' ? '' : '100px');
```

Comment:
Weird to go with the negation condition, but it's fine.
```suggestion
        this.$target.css('background-size', widgetValue === 'repeat-pattern' ? '100px' : '');
```